packages/nextjs/services/web3/connectors.tsx

```diff
@@ -23,12 +23,13 @@ function withDisconnectWrapper(connector: InjectedConnector) {
 function getConnectors() {
   const { targetNetworks } = scaffoldConfig;
 
-  const connectors = [argent(), braavos()];
+  const connectors: InjectedConnector[] = [argent(), braavos()];
 
   if (
     targetNetworks.some((network) => (network.network as string) === "devnet")
   ) {
-    connectors.push(new BurnerConnector());
+    const burnerConnector = new BurnerConnector();
+    connectors.push(burnerConnector as unknown as InjectedConnector);
```

Comment:
just curious what is the reason to refactor it like this?

Reply:
The root cause of this issue lies in the fact that `InjectedConnector` base class from `@starknet-react/core` has a private `_wallet` property, while `BurnerConnector` as a subclass doesn't explicitly define this property.
In TypeScript, when a class inherits from another class, if the parent class has private properties, the child class must somehow "know" about the existence of these private properties. Due to the scope restrictions of private properties, these properties must come from the same declaration context (same package/module).

Reply:
i see, thanks!